APE24.rst

```diff
@@ -0,0 +1,497 @@
+
+pyOpenSci will respect the Astropy-specific criteria for someone to be a
+reviewer for Affiliated request, namely:
+
+* familiarity with the Astropy project,
+* ability to judge whether a package integrates well with the Astropy ecosystem
+  (as per pre-APE 24 guidelines), and
+* having domain expertise in the area of the package (e.g., galaxy evolution).
+
+pyOpenSci and Astropy both ask editors/reviewers disclose any potential conflict of interest (COI)
+prior to agreeing to review a package. In the event where COI occurs, with this partnership,
+we would follow the
+`pyOpenSci COI process <https://www.pyopensci.org/software-peer-review/our-process/policies.html#conflict-of-interest-for-reviews-and-editors>`_.
+pyOpenSci invites the Astropy community to review this language to ensure it meets our needs.
+
+.. _reviewer sign-up form: https://forms.gle/GHfxvmS47nQFDcBM6
+
+Review turnaround time
+----------------------
+
+Astropy currently does not enforce any concrete turnaround time. Reviews typically
+come within weeks, but the response time for submitted packages to address the review
+is very non-uniform from "within days" to "years".
+
+If we partner with pyOpenSci, Astropy would need to adhere to their expected
+timeline for each step, as laid out in `An Overview Of the Peer Review Process`_.
+For example, editor is expected to find reviewers within 2-3 weeks and
+a peer review should be completed within 3 weeks after that.
+This is to ensure that the package maintainer have a good experience with the review
+and things do not languish over a long period of time.
+
+Generally, the editor role should not take a huge amount of time, *but* it is
+important for an editor, once the review starts, to check in on the review periodically
+(every few weeks and more often during wrap-up).
+
+.. _An Overview Of the Peer Review Process: https://www.pyopensci.org/software-peer-review/our-process/review-timeline.html
+
+
+Packaging guidelines
+--------------------
+
+The Astropy community has followed packaging guidelines published in the 
+`OpenAstronomy packaging guide`_ for a few years,
+and `Astropy package template <https://github.com/astropy/package-template>`_ before that.
+
+pyOpenSci is also developing a `community-driven packaging guide`_
+that covers modern best practices and recommendations for scientific Python packaging. 
+This guide has a stringent community review process. Reviewers for this guide
+represent members of PyPA, core Python, Anaconda (conda/conda-forge), and
+core packages for front- and back-end tools (e.g., ``flit``, PDM, ``hatch``/``hatchling``, etc.).  
+
+If we partner with pyOpenSci, we encourage our Affiliated packages to continue to follow the
+`OpenAstronomy packaging guide`_ for packaging structure. However maintainers will find the
+pyOpenSci guide to be a useful reference, given it provides community-wide guidelines and
+is targeted to those newer to packaging.
```

Comment:
```suggestion
`OpenAstronomy packaging guide`_ for packaging structure, which ensures a packaging
style consistent with the majority of existing Affiliated packages as well as large parts of
the greater scientific Python environment, in particular Numpy and Scipy.
However maintainers will find the pyOpenSci guide to be a useful reference, given it provides
community-wide guidelines and is targeted to those newer to packaging.
Editors and reviewer shall not let a particular choice of packaging influence their rating of
a package.
```
some rationale and clarification based on the discussion on @Cadair's comments.

Reply:
Actually this may warrant some further clarification in that we will require _some_ sort of packaging structure.
E.g. we have just recently included having a `CITATION` file in the requirements for new Affiliated packages, and should continue to ask for something with that function, though we would not demand it is called exactly that, or where in the repository it is sitting.

Reply:
@dhomeier i really appreciate these edits but i'd love to chat a bit more about the language. 

> for packaging structure, which ensures a packaging style consistent with the majority of existing Affiliated  packages as well as large parts of the greater scientific Python environment, in particular Numpy and Scipy.

as i really worry it will confuse users. It is true that all of the core packages - numpy, scipy etc. use a flat layout! i actually did a survey of the ecosystem. but it's not true that the greater scientific Python ecosystem embraces it. 

There are many packages using the src/ layout now. in fact the entire scikit-hep community (and [this is now in the scientific-python.org packaging guide which really represents those core packages](https://learn.scientific-python.org/development/tutorials/packaging/)). As such what do you think about loosening up this language a bit?  

Maybe something like 

> packaging
style consistent with the majority of existing astropy Affiliated packages as well as the core scientific Python packages such as Numpy and Scipy.  

instead of :
> as well as large parts of
the greater scientific Python environment, in particular Numpy and Scipy.

i just worry about our users getting confused. because scientific-python.org, pypa, setuptools etc all are promoting src/ layout in their docs at this point. so i'd rather see more relaxed language. i welcome your thought and really understand and appreciate where you are coming from.

Reply:
I am entirely open to rewording, and happy to leave it to native speakers to iron out the fine print.
Regarding content, I think from https://github.com/astropy/astropy-APEs/pull/87#discussion_r1360898262 there really is consensus that we don't want to force a package to change their existing and functional layout.
But I doubt there is as broad support for recommending the scientific-python.org guide over the Openastronomy one for packages just setting their structure up – think it really comes down to balancing potential ease of configuration against joining or staying in a familiar environment.
For me personally I'll be much more involved in working with the code of other Astropy-Affiliated packages or the core scientific Python system (agreed that is a much better description for Numpy, Scipy etc.) than dealing with the internals of pypa, setuptools or even the scikit family. But that will obviously vary a lot individually.

Reply:
I suggest that we worry too much about the language here. This is an APE where we propose to change our process, it's not the help page that users will see in in the docs. 
We also all agree that different layouts are possible, so I suggest we lead with that upfront. Something like

> There are different standards in the Python community for how to lay out a package and we do not force any particular choice. However, we encourage to follow  ... openastronomy ... for consistency ...

Reply:
this all sounds wonderful to me!! @dhomeier i hear you!! and @hamogu i like your suggested wording. and got it - it's just an APE! i'm just hoping we can use language throughout our scientific ecosystem that helps users understand there are options and it's ok if they have already selected an option if it is working for them. i just know ppl get confused from talking to them BUT like you said this APE isn't necessarily public facing documentation  ! i think we have consensus here.

Reply:
Changed some of the wording according to the first comments

Reply:
Okay, given that Derek has edited his suggestion above to account for the comments here, I am going to accept it. Thanks!